File: c++/bubbleSort.cpp
#include <iostream>
#include <stdlib.h>
#include <time.h> 
using namespace std;

int bubble_sort(int *arr, int size);

int main()
{
    int size = 10;
    int *myArr = new int[size];
    srand (time(NULL));

    for (int i = 0; i < size; i += 1)
    {
        myArr[i] = rand() % 101 + -100;
    }

    *myArr = bubble_sort(myArr, size);

    for (int i = 0; i < size; i += 1)
    {
        cout << myArr[i] << endl;
    }

    return 0;
}

int bubble_sort(int *arr, int size)
{
    bool sorted = false;
    while (!sorted)
    {
        sorted = true;
        for (int i = 0; i < size - 1; i += 1)
        {
            if (arr[i] > arr[i + 1])
            {
                sorted = false;
                arr[i] = arr[i] ^ arr[i + 1];
                arr[i + 1] = arr[i] ^ arr[i + 1];
                arr[i] = arr[i] ^ arr[i + 1];
            }
        }
    }

    return *arr;
}

File: go/bubbleSort.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func Bubble_sort(lst []int) []int {
	var sorted = false
	for !sorted {
		sorted = true
		for i := 0; i < len(lst)-1; i++ {
			if lst[i] > lst[i+1] {
				sorted = false
				lst[i] = lst[i] ^ lst[i+1]
				lst[i+1] = lst[i] ^ lst[i+1]
				lst[i] = lst[i] ^ lst[i+1]
			}
		}
	}
	return lst
}

func main() {
	const size = 10
	var arr []int
	s1 := rand.NewSource(time.Now().UnixNano())
	r1 := rand.New(s1)
	for i := 0; i < size; i++ {
		arr[i] = r1.Intn(100)
	}
	fmt.Println(Bubble_sort(arr))
}
